react-native/setup/setup-firebase.mo.md

```diff
@@ -0,0 +1,140 @@
+
+* At the top of the file:
+```c
+#import <Firebase.h>
+```
+
+* In the `didFinishLaunchingWithOptions:(NSDictionary *)launchOptions` method before the return:
+```c
+[FIRApp configure];
+```
+
+On the Podfile `ios/Podfile`, check the platform version at least '9.0':
+
+`platform :ios, '9.0'`
+
+and add the requirement:
+```
+pod 'Firebase/Core'
+pod 'Firebase/Analytics'
+```
+
+run `pod install` from your `ios/` folder.
+
+### Setup Android *(~8 min)*
+#### Setup google-services.json
+Download the `google-services.json` from the firebase console and paste it on `android/app/google-services.json`.
+
+Add the folowing lines on the `android/build.gradle` file:
+```gradle
+buildscript {
+  // ...
+  dependencies {
+    // ...
+    classpath 'com.google.gms:google-services:3.1.1'
+  }
+}
+```
+
+and at the VERY BOTTOM of the `android/app/build.gradle` add:
+`apply plugin: 'com.google.gms.google-services'`
+
+#### Add Firebase modules
+On the `android/app/build.gradle` add:
+```gradle
+dependencies {
+  // This should be here already
+  compile(project(':react-native-firebase')) {
+    transitive = false
+  }
+
+  // Firebase dependencies
+  compile "com.google.android.gms:play-services-base:11.4.2"
+  compile "com.google.firebase:firebase-core:11.4.2"
+  compile "com.google.firebase:firebase-analytics:11.4.2"
+
+  ...
+```
+
+and add the following lines to `android/build.gradle`:
+```gradle
+allprojects {
+    repositories {
+        mavenLocal()
+        jcenter()
+        maven {
+            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+            url "$rootDir/../node_modules/react-native/android"
+        }
+        // -------------------------------------------------
+        // Add this below the existing maven property above
+        // -------------------------------------------------
+        maven {
+            url 'https://maven.google.com'
+        }
+    }
+}
+```
+
+Update you `android/app/src/main/java/com/[app name]/MainApplication.java` to integrate the Firebase analytics package:
+
+```java
+// ...
+import io.invertase.firebase.RNFirebasePackage;
+import io.invertase.firebase.analytics.RNFirebaseAnalyticsPackage; // <-- Add this line
+
+public class MainApplication extends Application implements ReactApplication {
+    // ...
+
+    @Override
+    protected List<ReactPackage> getPackages() {
+      return Arrays.<ReactPackage>asList(
+          new MainReactPackage(),
+          new RNFirebasePackage(),
+          new RNFirebaseAnalyticsPackage() // <-- Add this line
+      );
+    }
+  };
+  // ...
+}
+```
+
+
+### Check your google services version *(~10 min)*
+
+The google services version used in Firebase could be in conflict with the version of other native modules (as react-native-maps).
+
+The following [Medium blog post](https://medium.com/@suchydan/how-to-solve-google-play-services-version-collision-in-gradle-dependencies-ef086ae5c75f) propose a way to maintain the same version.
```

Comment:
This post is good in this particular case, if you want to have a deeper understanding of why you have an error - which might be required with the next versions - I have a dev standard in PR https://github.com/bamlab/dev-standards/pull/66 ;)

Reply:
I've seen it :D This standard is perfect to explain this issue. I'll link it here.